cpp/include/cudf/io/functions.hpp

```diff
@@ -769,8 +769,16 @@ void write_parquet_chunked(table_view const& table,
  *
  * @param[in] state Opaque state information about the writer process. Must be the same pointer
  * returned from write_parquet_chunked_begin()
+ * @param[in] return_filemetadata If true, return the raw file metadata
+ * @param[in] metadata_out_file_path Column chunks file path to be set in the raw output metadata
+ *
+ * @return A blob that contains the file metadata (parquet FileMetadata thrift message) if
+ *         requested in write_parquet_args (empty blob otherwise)
  */
-void write_parquet_chunked_end(std::shared_ptr<detail::parquet::pq_chunked_state>& state);
+std::unique_ptr<std::vector<uint8_t>> write_parquet_chunked_end(
```

Comment:
I'm more of a fan of "pass a pointer that defaults to null for optional outputs" for this kind of thing, but I guess the top level interface already does it this way.